eclipse-scout-core/src/testing/JasmineScoutUtil.ts

```diff
@@ -130,5 +130,90 @@ export const JasmineScoutUtil = {
         });
       }
     });
+  },
+
+  /**
+   * Asserts that every page has an uuid and a specific {@link PageParamDo}, if required.
+   */
+  assertPageCompleteness(options?: PageCompletenessOptions) {
+    options = options || {};
+    let pagesNotRequiringUuid: Set<Constructor<Page> | AbstractConstructor<Page>> = new Set([PageWithNodes, PageWithTable, AutoLeafPageWithNodes, ...options.pagesNotRequiringUuid || []]);
+    let pagesNotRequiringPageParam: Set<Constructor<Page> | AbstractConstructor<Page>> = new Set([PageWithNodes, PageWithTable, AutoLeafPageWithNodes, ...options.pagesNotRequiringPageParam || []]);
+    let missingUuids = new Set();
+    let missingPageParams = new Set();
+    let completePages = new Set();
+
+    for (const PageConstructor of ObjectFactory.get().getSubClassesOf(Page)) {
+      let pageType = ObjectFactory.get().getObjectType(PageConstructor);
+      if (options.namespace && !pageType.startsWith(options.namespace)) {
+        continue;
+      }
+
+      let page = new PageConstructor();
+      page.minimalInit();
+
+      // Assert uuid
+      if (scout.nvl(options.assertUuid, true) && !page.uuid && !pagesNotRequiringUuid.has(PageConstructor)) {
+        missingUuids.add(pageType);
+      }
+
+      // Assert pageParam
+      if (scout.nvl(options.assertPageParam, true)) {
+        let pageParamType = `${pageType}ParamDo`;
+        let PageParam = TypeDescriptor.resolveType(pageParamType);
+        if (!PageParam && !pagesNotRequiringPageParam.has(PageConstructor)) {
+          missingPageParams.add(pageType);
+        }
+      }
+
+      if (!missingUuids.has(pageType) && !missingPageParams.has(pageType)) {
+        completePages.add(pageType);
+      }
+    }
+
+    if (missingUuids.size > 0) {
+      fail(`Found page(s) without a uuid. Please ensure every page has a uuid.
+      ${Array.from(missingUuids).join('\n')}`);
```

Comment:
The second line of this string has leading spaces, is this intended?

The syntax for multiline strings is a bit ugly. Do we really need them? I think classic strings would work better here. Or something like this:

```
fail([
  `Found ${missingUuids.length} pages without a uuid. Please ensure every page has a uuid.`,
  ...missingUuids
].join(\n'));
```